### PR TITLE
Delete unnecessary caches

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/Common/src/CoreLib/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs [hunk 1]
```diff
@@ -11,17 +11,14 @@ namespace Microsoft.Win32.SafeHandles
 {
     public sealed class SafeFileHandle : SafeHandleZeroOrMinusOneIsInvalid
     {
-        /// <summary>A handle value of -1.</summary>
-        private static readonly IntPtr s_invalidHandle = new IntPtr(-1);
```
Comment (jkotas) [inline src/Common/src/CoreLib/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs]: We use inline more efficient `new IntPtr(-1)` in most places. There were a few places caching it for no good reason.

File: src/System.Console/src/System/ConsolePal.Windows.cs [hunk 2]
```diff
@@ -14,11 +14,9 @@ internal static class ConsolePal
     {
         private const int DefaultConsoleBufferSize = 256; // default size of buffer used in stream readers/writers
 
-        private static IntPtr s_InvalidHandleValue = new IntPtr(-1);
+        private static IntPtr InvalidHandleValue => new IntPtr(-1);
 
-        private static bool s_isWindows7 = GetIsWindows7();
```
Comment (jkotas) [inline src/System.Console/src/System/ConsolePal.Windows.cs]: I have noticed initialization of `s_isWindows7` running during a simple `Console.WriteLine("Hello world")`. The OS version is cached by Environment already, and IsWindows7 check is on rare path in Title property. It is fine better to just recompute it each time it is needed without caching.

File: src/System.Console/src/System/ConsolePal.Windows.cs [hunk 3]
```diff
@@ -14,11 +14,9 @@ internal static class ConsolePal
     {
         private const int DefaultConsoleBufferSize = 256; // default size of buffer used in stream readers/writers
 
-        private static IntPtr s_InvalidHandleValue = new IntPtr(-1);
+        private static IntPtr InvalidHandleValue => new IntPtr(-1);
```
Comment (danmosemsft) [inline src/System.Console/src/System/ConsolePal.Windows.cs]: So worth keeping this one?

Comment (jkotas) [reply src/System.Console/src/System/ConsolePal.Windows.cs]: This one was used in a number of places so I have changed it to property that will get inlined. It is no longer changed in a static.

**Comment by jkotas** [comment]
@dotnet-bot test Windows x86 Release Build please

**Pull request merged**